nifi-toolkit/nifi-toolkit-assembly/docker/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements. See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership. The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM openjdk:8-jre-alpine
+LABEL maintainer "Apache NiFi <dev@nifi.apache.org>"
+
+ARG UID=1000
+ARG GID=1000
+ARG NIFI_TOOLKIT_VERSION=1.4.0-SNAPSHOT
```

Comment:
Will hardcoding the version here (and in Dockerfile.hub) cause extra maintenance for releases after 1.4.0?

Reply:
-SNAPSHOT should be removed because it's a mavenism that applies to non-release builds.

Reply:
Using this approach might be useful for injecting the build values into the Dockerfile prior to it being processed with Docker: https://stackoverflow.com/questions/34289889/how-to-inject-maven-profile-value-into-properties-file#34290331

Reply:
@MikeThomsen Will evaluate the latter but given how the structure of docker images is currently composed this is needed as it will be making use of the generated nifi toolkit as part of the build which will be a SNAPSHOT.

The Dockerfile.hub version is devoid of this given that it only is built against generated releases.

Reply:
AFAIK, Maven provides a build variable you can use for that. So you should be able to collapse the two together at some point if you take advantage of that assuming that there are no other reasons to keep them separate.

Reply:
Updated with a new commit that does resource filtering as linked above.